src/lib.rs

```diff
@@ -19,12 +19,22 @@ pub mod worktree;
 /// The bool in the return value specifies whether there is a repository
 /// in root itself.
 #[allow(clippy::type_complexity)]
-fn find_repos(root: &Path) -> Result<Option<(Vec<repo::Repo>, Vec<String>, bool)>, String> {
+fn find_repos(
+    root: &Path,
+    exclusion_pattern: Option<&str>,
+) -> Result<Option<(Vec<repo::Repo>, Vec<String>, bool)>, String> {
     let mut repos: Vec<repo::Repo> = Vec::new();
     let mut repo_in_root = false;
     let mut warnings = Vec::new();
 
+    let exlusion_regex: regex::Regex = regex::Regex::new(&exclusion_pattern.unwrap_or(r"^$"))
```

Comment:
Clippy complains about this (this expression creates a reference which is immediately dereferenced by the compiler), and want to see this changed to the following:

```suggestion
    let exlusion_regex: regex::Regex = regex::Regex::new(exclusion_pattern.unwrap_or(r"^$"))
```

Short explanation: The `unwrap_or()` will already return a `&str` (as this is the content of the `Option`). There is no point in passing a reference of a `&str` to `Regex::new()`. This would be `&&str`. So in this case, the `&` is not needed.